compiler/passes/convert-uast.cpp

```diff
@@ -2916,9 +2916,9 @@ struct Converter {
       if (varLikeDecl == nullptr) continue;
       if (varLikeDecl->storageKind() != types::QualifiedType::TYPE) continue;
 
-      auto assocTypeName = varLikeDecl->name().c_str();
+      auto assocTypeName = varLikeDecl->name();
       noteConvertedSym(varLikeDecl,
-                       isymAssociatedTypes.at(assocTypeName)->symbol);
+                       isymAssociatedTypes.at(assocTypeName.c_str())->symbol);
```

Comment:
IMO it would be nice if `std::map<const char*, ConstrainedType*, CharStarComparator> associatedTypes` moved to `std::map<UniqueString, ConstrainedType*> associatedTypes` but obviously such a change might have other impacts. (And we have other similar issues in the production AST in other places).